client/lecturer/app/engagement.directive.js

```diff
@@ -19,9 +19,32 @@
         function link(scope, element, attrs, engagementCanvas) {
             console.log('Ready (Engagement Canvas link)');
 
-            engagementCanvas.canvas = element[0];
+            engagementCanvas.width = element.width();
+            engagementCanvas.height = element.height();
+
+            engagementCanvas.canvas = document.createElement('canvas');
+            element.append(engagementCanvas.canvas);
             engagementCanvas.ctx = engagementCanvas.canvas.getContext('2d');
 
+            engagementCanvas.canvas.width = engagementCanvas.width;
+            engagementCanvas.canvas.height = engagementCanvas.height;
```

Comment:
Well,  both canvas should be the same size so why have two separate variables for them?
if `engagementCanvas.width` makes more sense, then you can rename the other...

I do not see any point in renaming `element` though.